tests/src/performance/JitBench/JitBenchHarness.cs

```diff
@@ -0,0 +1,355 @@
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            var options = JitBenchHarnessOptions.Parse(args);
+
+            s_temporaryDirectory = Path.Combine(options.IntermediateOutputDirectory, "JitBenchHarness");
+            if (Directory.Exists(s_temporaryDirectory))
+                Directory.Delete(s_temporaryDirectory, true);
+            Directory.CreateDirectory(s_temporaryDirectory);
+
+            s_jitBenchDevDirectory = Path.Combine(s_temporaryDirectory, "JitBench-dev");
+
+            using (var h = new XunitPerformanceHarness(args))
+            {
+                var startInfo = Setup();
+                PrintHeader("Running Benchmark Scenario");
+                h.RunScenario(startInfo, () => { }, PostIteration, PostProcessing, s_ScenarioConfiguration);
+            }
+        }
+
+        static Program()
+        {
+            s_ScenarioConfiguration = new ScenarioConfiguration(TimeSpan.FromMilliseconds(20000)) {
+                Iterations = 11
+            };
+
+            // Set variables we will need to store results.
+            s_iteration = 0;
+            s_startupTimes = new double[s_ScenarioConfiguration.Iterations];
+            s_requestTimes = new double[s_ScenarioConfiguration.Iterations];
+        }
+
+        private static ProcessStartInfo Setup()
+        {
+            PrintHeader("Starting SETUP");
+
+            var dotnetDirectory = Path.Combine(s_jitBenchDevDirectory, ".dotnet");
+            var dotnetProcessFileName = Path.Combine(dotnetDirectory, "dotnet.exe");
+
+            //Download and extract the repo.
+            using (var client = new HttpClient())
+            {
+                // Download the JitBench repository and extract it.
+                var url = @"https://github.com/guhuro/JitBench/archive/dev.zip"; // TODO: This should be updated to https://github.com/aspnet/JitBench/archive/dev.zip once the fork is merged.
+                var zipFile = Path.Combine(s_temporaryDirectory, "dev.zip");
+
+                using (var tmpzip = File.Create(zipFile))
+                {
+                    using (var stream = client.GetStreamAsync(url).Result)
+                        stream.CopyTo(tmpzip);
+                    tmpzip.Flush();
+                }
+
+                // If the repo already exists, we delete it and extract it again.
+                if (Directory.Exists(s_jitBenchDevDirectory))
+                    Directory.Delete(s_jitBenchDevDirectory, true);
+
+                // This step will create s_JitBenchDevDirectory.
+                ZipFile.ExtractToDirectory(zipFile, s_temporaryDirectory);
+            }
+
+            // first dotnet-install.ps1
+            var installProcessStartInfo = new ProcessStartInfo() {
+                WorkingDirectory = s_jitBenchDevDirectory,
+                FileName = @"powershell.exe",
+                Arguments = @".\Dotnet-Install.ps1 -SharedRuntime -InstallDir .dotnet -Channel master -Architecture x64"
```

Comment:
We should ensure that these are still the instructions for running the JitBench scenario.  I know there has been some churn on how the thing runs.